### DISCUSSION TITLE
Allow manually configure wireguard fwmark (in linux at least) and maybe - maybe a custom ip rule section

### DISCUSSION THREAD

**Discussion by LindaFerum** [mullvad/mullvadvpn-app#9082] [Feature requests & Ideas]
## Issue report

Operating system: All linuxes

App version: a future one I guess because it's a feature request

### Issue description

Basically, allow mullvad app users to specify a custom fwmark (somewhere in advanced options) for wireguard 

Wireguard already has this power "under the hood" so it's mostly down to providing a UI facility in the app for setting the fwmark to what user wants

ideally also maybe a UI place (or even just a user-editable config somewhere, it's "advanced power user" stuff after all) that allows to override ip routing rules being used or add one's own 

But even just fwmark configuration alone would enable some very neat advanced usecases and is probably not a very huge burden in terms of implementing

**Comment by pinkisemils** [comment]
We will consider this a feature request. We haven't even considered allowing users to change the firewall mark we use for routing and blocking traffic. If you want to augment the firewall or routing behavior in the mean time, you may use our firewall mark verbatim - [0x6d6f6c65](https://github.com/mullvad/mullvadvpn-app/blob/main/mullvad-types/src/lib.rs#L24). We do mention that users can use this firewall mark to exclude traffic via their own firewall rules if they so please in the [advanced split tunnelling guide](https://mullvad.net/en/help/split-tunneling-with-linux-advanced/).
